android-sdk-simplexml/src/main/java/com/microsoft/hsg/android/simplexml/client/HealthVaultRestClient.java

```diff
@@ -0,0 +1,164 @@
+import retrofit2.converter.gson.GsonConverterFactory;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+import retrofit2.http.Body;
+import retrofit2.http.GET;
+import retrofit2.http.HTTP;
+import retrofit2.http.Headers;
+import retrofit2.http.PATCH;
+import retrofit2.http.POST;
+import retrofit2.http.PUT;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+
+public class HealthVaultRestClient implements IHealthVaultRestClient {
+
+	private static MicrosoftHealthVaultRestApiImpl instance;
+	private static DateTime mLastRefreshedSessionCredential = DateTime.now();;
+	private static final int mSessionCredentialCallThresholdMinutes = 5;
+	private static Retrofit mRetrofit;
+	private static String mRestURL;
+	private static OkHttpClient.Builder mOkBuilder;
+	private static Connection mConnection;
+	private static Record mCurrentRecord;
+	private static HealthVaultSettings mSettings;
+
+	public HealthVaultRestClient (HealthVaultSettings settings, Connection connection, Record currentRecord){
+		Initialize(settings, connection, currentRecord);
+	}
+
+	private void Initialize(HealthVaultSettings settings, Connection connection, Record currentRecord){
+		if (connection == null){
+			throw new HVException("connection is null");
+		}
+		mSettings = settings;
+		mRestURL = settings.getRestUrl();
+		mCurrentRecord = currentRecord;
+		mConnection = connection;
+		mOkBuilder = getOkHttp(connection, currentRecord);
+		mRetrofit = getRetrofit(mRestURL);
+	}
+
+	public MicrosoftHealthVaultRestApiImpl getClient() {
+		return new MicrosoftHealthVaultRestApiImpl(mRestURL, mOkBuilder, mRetrofit.newBuilder());
+	}
+
+	private Retrofit getRetrofit(String url){
+		Retrofit retrofit = new Retrofit.Builder()
+				.baseUrl(url)
+				.addConverterFactory(GsonConverterFactory.create())
+				.addCallAdapterFactory(RxJavaCallAdapterFactory.create())
+				.build();
+		return retrofit;
+	}
+
+	public OkHttpClient.Builder getOkHttp(final Connection connection, final Record currentRecord){
+		OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder();
+		clientBuilder.connectionPool(new ConnectionPool(3, 5, TimeUnit.SECONDS));
+		clientBuilder.addInterceptor(new Interceptor() {
+			@Override
+			public okhttp3.Response intercept(Chain chain) throws IOException {
+				String token = getAuthToken(connection, currentRecord);
+				Request request = chain.request();
+				Request.Builder newRequest = request.newBuilder().header("Authorization", token);
+				newRequest.build();
+
+				return chain.proceed(newRequest.build());
+			}
+		});
+		return clientBuilder;
+	}
+
+	private String getAuthToken (Connection connection, Record currentRecord){
+		DateTime dateTime= mSettings.getSessionExpiration();
+		if(dateTime.isBeforeNow()) {
+			tokenRefresh(connection);
+		}
+
+		String token =  "MSH-V1 app-token=" + connection.getSessionToken().toString() +
+				",offline-person-id=" + currentRecord.getPersonId() +
+				",record-id=" + currentRecord.getId();
```

Comment:
Are we guaranteed that apptoken, offlinepersonid and recordid are available always? I am not sure if the service will accept a null or empty value for those. You probably should not concat those if they are not available. 

Also, consider using StringBuilder instead of String concat.